src/routes/users/sign-up-route.js

```diff
@@ -1,37 +1,72 @@
-const qs = require('querystring');
+const fs = require("fs");
+const path = require("path");
 
 const saveUser = user => {
-  // получить файл с юзером
-  // найти путь папки users
-  // сохранить туда файл
+  const userName = user.username;
+  const fileUser = path.join(
+    __dirname,
+    "../../",
+    "db",
+    "/users",
+    `${userName}.json`
+  );
+  fs.writeFile(fileUser, JSON.stringify(user), function(err) {
+    if (err) throw err;
+    console.log(`${userName}.json was created`);
+  });
 };
 
-const signUpRoute = (request, response) => {
-  // Взять данные что пришли
+const checkUser = user => {
+  const userName = user.username;
+  const userPhone = user.telephone.replace(/\s/g, "");
+  const userPass = user.password;
+  const userEmail = user.email;
+  if (
+    typeof userName === "string" &&
+    !isNaN(Number(userPhone)) &&
+    typeof userPass === "string" &&
+    typeof userEmail === "string"
+  )
+    return true;
+  else return false;
+};
 
-  if (request.method === 'POST') {
-    let body = '';
+const signUpRoute = (request, response) => {
+  if (request.method === "POST") {
+    let body = "";
 
-    request.on('data', function (data) {
+    request.on("data", function(data) {
       body = body + data;
-
-      console.log('Incoming data!!!!');
-    });
-
-    request.on('end', function () {
-      const post = qs.parse(body);
-      console.log(post);
+      console.log("Incoming data!!!!");
     });
-  }
-
-  // Взять username с данных, сохранить в переменную
 
-  // Сохраняем данные в <username>.json
+    request.on("end", function() {
+      const user = JSON.parse(body);
+      if (checkUser(user)) {
```

Comment:
Also you should check if user with such name already exists. If so - throw error. Since your current implementation will overwrite existing user with new!!!